vtex-sites/nextjs.store
# chore: Sync search input changes

## What's the purpose of this pull request?

While porting [Gatsby's feat: Add search history & top searches #67](https://github.com/vtex-sites/gatsby.store/pull/67) to this repo I noticed we had some other changes that needed to be ported first. Gatsby's version is the most up-to-date one and should be used as the base. 

In this PR I port the remaining changes to help to keep both repositories in sync, as much as possible (there are some minor differences intrinsic to the underlying framework).

## How does it work?

I had first to [undo the use of CSS Modules](https://github.com/vtex-sites/nextjs.store/commit/a6ce7c537b3a27e160b6f8b5f43daa253735ae78) for the `Suggestions` and `SuggestionProductCard` components because Gatsby's doesn't use it **yet**. I had to use CSS Modules to address some unused CSS that was blocking https://github.com/vtex-sites/nextjs.store/pull/33. Undoing it now was safe because we have other components using CSS modules so our unused CSS is below enough the threshold.

I then proceeded to apply this refactoring from [Gatsby's chore: Organize some search components #7](https://github.com/vtex-sites/gatsby.store/pull/7).

## How to test it?

Visit the preview link and test the search input. The current version is limited to showing only suggested terms and products. There should be no visual or functional regression from the production's version.
